src/app/static/src/app/components/plots/bubble/BubblePlot.vue

```diff
@@ -207,23 +210,54 @@
                 return unsetFilters.length == 0 && this.selections.detail > -1 &&
                     !!this.selections.colorIndicatorId && !!this.selections.sizeIndicatorId;
             },
-            indicatorRanges() {
-                return getIndicatorRanges(this.chartdata, this.indicators)
-            },
             currentLevelFeatureIds() {
                 return this.currentFeatures.map(f => f.properties!!["area_id"]);
             },
-            colourRanges() {
-                let selectedCurrentLevelAreaIds = this.selectedAreaIds.filter(a => this.currentLevelFeatureIds.indexOf(a) > -1);
-
-                return getColourRanges(
-                    this.chartdata,
-                    this.indicators,
-                    this.colourScales || {},
-                    this.nonAreaFilters,
-                    this.selections.selectedFilterOptions,
-                    selectedCurrentLevelAreaIds
-                )
+            sizeRange() {
+                if (!this.initialised) {
+                    return {max: 1, min: 0}
+                }
+                const sizeId = this.selections.sizeIndicatorId;
+                if (!this.fullIndicatorRanges.hasOwnProperty(sizeId)) {
+                    // cache the result in the fullIndicatorRanges object for future lookups
+                    this.fullIndicatorRanges[sizeId] =
+                        getIndicatorRange(this.chartdata, this.sizeIndicator)
+                }
+                return this.fullIndicatorRanges[sizeId];
+            },
+            colourRange() {
+                const colorId = this.selections.colorIndicatorId;
```

Comment:
This logic needs to live in the component itself so that the result is cached and it doesn't get re-calculated unnecessarily. To tidy up these 2 components which are massive and share a lot of code, we should refactor to make use of mixins. I'll open a ticket to do this.

Reply:
Good idea, wanted to do that anyway, even more desirable now!

Reply:
https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-1448